packages/nx/src/config/project-graph.ts

```diff
@@ -107,13 +107,16 @@ export interface ProjectGraphProjectNode<T = any> {
 
 /**
  * A node describing an external dependency
+ * `name` has as form of `npm:packageName` for primary dependencies or `npm:packageName@version` for hoisted ones
  */
 export interface ProjectGraphExternalNode {
   type: 'npm';
   name: `npm:${string}`;
   data: {
     version: string;
     packageName: string;
+    dependencies?: Record<string, [string, string]>; // dependencies of this version { [packageName]: [versionRange, actualVersion] }
```

Comment:
Why are dependencies here instead of being in the dependencies section of the graph?

Reply:
There is no good reason I guess. I will move it to dependencies and return a partial graph from the mapper instead of the external Dependencies only.